packages/react/xr/src/origin.tsx

```diff
@@ -4,29 +4,38 @@ import { Group } from 'three'
 import { xrSpaceContext } from './contexts.js'
 import { useXR } from './xr.js'
 
+export type XROriginProps = ThreeElements['group'] & {
+  disabled?: boolean
+}
+
 /**
  * Component for setting the origin of the player (their feet)
  *
  * @param props
  * Accepts the same props as a ThreeJs [Group](https://threejs.org/docs/#api/en/objects/Group)
  * @function
  */
-export const XROrigin = forwardRef<Group, ThreeElements['group']>(({ children, ...props }, ref) => {
+export const XROrigin = forwardRef<Group, XROriginProps>(({ children, disabled, ...props }, ref) => {
   const xrCamera = useThree((s) => s.gl.xr.getCamera())
   const internalRef = useRef<Group>(null)
-  useImperativeHandle(ref, () => internalRef.current!, [])
   const referenceSpace = useXR((xr) => xr.originReferenceSpace)
+
+  useImperativeHandle(ref, () => internalRef.current!, [])
+
   useEffect(() => {
     const group = internalRef.current
-    if (group == null) {
+    if (!group || disabled) {
```

Comment:
I might have overlooked this change or it is new but I also prefer explicit null comparisons.

Side question: did you manually re-order the hooks or is that something automatic?

Reply:
😂